src/setup_payload/java/src/chip/setuppayload/SetupPayloadParser.java

```diff
@@ -4,14 +4,39 @@
 public class SetupPayloadParser {
 
   /** Returns {@link SetupPayload} parsed from the QR code string. */
-  public SetupPayload parseQrCode(String qrCodeString) throws UnrecognizedQrCodeException {
-    return fetchPayloadFromQrCode(qrCodeString);
+  public SetupPayload parseQrCode(String qrCodeString)
+      throws UnrecognizedQrCodeException, SetupPayloadException {
+    return fetchPayloadFromQrCode(qrCodeString, false);
+  }
+
+  /**
+   * Returns {@link SetupPayload} parsed from the QR code string.
+   *
+   * @param qrCodeString the QRCode for commissioning device.
+   * @param allowInvalidPayload Allow invalid payload components. If this value is true, payload
+   *     element validation is not checked. (See detail 5.1.3 in Matter Spec.)
```

Comment:
Please don't force me to just read the 5.1.3. Documentation that just says "read another documentation" is not friendly.

It is ok to point to it for full reference, but give me a TLDR variant here. Like packed binary structure should contain things like version,vendor and product id. This will not validate it.

Even after reading 5.1.3 I still do not know what `parseQrCode` will do with an invalid payload:
  - if the payload is invalid, what gets put in SetupPayload? Empty? Just accepting invalid values?
  - what is the element validation? Version? Range for vendor/product? Others?
  - will this even accept overly short payloads? what if a payload is 8 bits - 3 are for version 5 for vendor (?) ... is that even valid? what gets placed into vendor?
  
  Generally for me a "allow invalid data" with a returned parsed content sounds very vague and unclear. Unless SetupPayload does not contain the Onbarding payload bits.
  
  Please explain in the documentation.

Reply:
1. Added a phrase that an exception is throwing when an invalid element is included.
2. Content about invalid element cannot be written here, so it has been added to SetupPayload.java. And, added a phrase to refer to SetupPayload in the API document.